package.json

```diff
@@ -13,6 +13,7 @@
   ],
   "license": "MIT",
   "dependencies": {
+    "bluebird": "^3.5.1",
```

Comment:
Added a new dependency but at least it's the same as used in `serverless` itself

Reply:
I'd previously avoided using this as I saw no reason to include it.
However, the map and concurrency features look interesting...

Reply:
I think the old code had a bug as it wasn't properly awaiting the results of `this.provider.request` and had uncontrolled parallel uploads. And I really, really hate using Promises manually 😆

Reply:
The new code should run 3 "assets" in parallel while uploading "files" serially. So we have at maximum three uploads in parallel which should work well on most networks. Another change I would suggest is to switch from `putObject` to `upload` as I found it more robust and reliable. Not in this PR though.